web/modules/custom/joinup_core/joinup_core.module

```diff
@@ -869,41 +869,23 @@ function joinup_core_rdf_entity_update(EntityInterface $entity) {
  * published, the creation date is updated with the current datetime.
  */
 function joinup_core_node_presave(EntityInterface $entity) {
-  // Documents do not show the created date in the UI as they have the document
-  // publication date to show instead. However, since the listing screens sort
-  // by created date, the documents are updated just as the rest of the
-  // community content.
-  $community_bundles = ['discussion', 'document', 'event', 'news'];
-  if ($entity->isNew() || !in_array($entity->bundle(), $community_bundles) || !$entity->isPublished()) {
-    return;
+  // By default, publication_date module will set the value of `published_at` to
+  // `PUBLICATION_DATE_DEFAULT` value when the entity is not published and there
+  // is no value set for the field. According to publication_date, this is so
+  // that these entities come first in the results. Set this to NULL in this
+  // case.
```

Comment:
No results. We do not list unpublished items according to the publication date so normally this should not be a case.
However, I do not really like the `PUBLICATION_DATE_DEFAULT`. You claim it is an issue that the date NULL might be interpreted as 0 (January 1st, 1970) but I doubt it will be better to be interpreted as `Monday 18th January 2038 07:14:07 PM` my local time. How that would make it better? In the end we always check something. We either going to do a `if (empty($node->published_at->value)) {` or a `if ($node->published_at->value === PUBLICATION_DATE_DEFAULT) {`.

There is also the `@note: This is going to trigger the Year 2038 problem.` which I don't think it will happen but yeah, that. 

Another think I have an issue with the `PUBLICATION_DATE_DEFAUL` is that it *always applies*. If we follow the `\Drupal\publication_date\Plugin\Field\FieldType\PublicationDateItem::preSave` (remove my workaround), it will *always* set a value. They even overwrite the `::isEmpty` method by always returning `FALSE` (which yeah, it will also make NULL return a false `FALSE`). 
From the side of the developer, I find it more possible to think that an initially unpublished entity will have an empty publication value instead of an arbitrary value far in the future so that unpublished entities comes up first (why first? why should they come first?).

Reply:
These are points that should be raised in the upstream module. They could e.g. provide a pluggable mechanism for detecting unpublished content. I don't think this is something that should be overridden on our side, we should just integrate the functionality of the module.

Reply:
I created an issue upstream to address the `PUBLICATION_DATE_DEFAULT` problem: https://www.drupal.org/project/publication_date/issues/3066446